karaf/osgi_ds/osgi_ds_consumer/src/main/java/com/redhat/consulting/fusequickstarts/karaf/osgi_ds/route/ConsumerRoute.java

```diff
@@ -0,0 +1,92 @@
+package com.redhat.consulting.fusequickstarts.karaf.osgi_ds.route;
+
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.apache.camel.impl.SimpleRegistry;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.consulting.fusequickstarts.karaf.osgi_ds.api.IHelloService;
+
+/*
+ * This route is triggered by a Timer and called the Hello Service that is exposed from the other bundle.
+ */
+
+@Component(name="com.redhat.consulting.fusequickstarts.karaf.osgi_ds.route")
+public class ConsumerRoute extends DefaultCamelContext {
```

Comment:
Extended the Camel context in order to gain access to the Camel context to set beans in the Camel registry. Will add some additional comments in code.

I could rename class to eliminate any potential further confusion